sqlglot/dialects/tsql.py

```diff
@@ -370,6 +370,17 @@ def _timestrtotime_sql(self: TSQL.Generator, expression: exp.TimeStrToTime):
     return sql
 
 
+def _parse_datetrunc(args: t.List) -> exp.TimestampTrunc:
+    unit = seq_get(args, 0)
+    this = seq_get(args, 1)
+
+    # Every date - time type in tsql resoves to DATETIME2
+    if isinstance(this, exp.Expression) and this.is_string:
+        this = exp.TimeStrToTime(this=this)
+
+    return exp.TimestampTrunc(unit=unit, this=this)
```

Comment:
Minor nit, let's go with `exp.TimestampTrunc(this=this, unit=unit)`, `this` is generally the first arg in the constructors